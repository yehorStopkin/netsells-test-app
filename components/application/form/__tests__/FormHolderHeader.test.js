const { shallowMount } = require('@vue/test-utils');
const FormHolderHeader = require('@/components/application/form/FormHolderHeader').default;

describe('FormHolderHeader', () => {
    getData = (currentStep = 0, filingOfPersonalInfo = 0, filingOfAdditionalInfo = 0, filingOfAdditionalFiles = 0, availableSteps = [0]) => {
        return {
            currentStep,
            filingOfPersonalInfo,
            filingOfAdditionalInfo,
            filingOfAdditionalFiles,
            availableSteps,
        };
    };

    test('component default', () => {
        const wrapper = shallowMount(FormHolderHeader, { propsData: { ...getData() } });
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();

        const rootElement = wrapper.findAll('.form-holder-header');
        expect(rootElement.length).toBe(1);
        
        const applicationSteps = wrapper.findAll('.form-holder-header__application-step');
        expect(applicationSteps.length).toBe(3);
    });

    test('selected step 1', () => {
        const wrapper = shallowMount(FormHolderHeader, { propsData: { ...getData(0) } });

        expect(wrapper.vm.personalInfoSelected).toBe(true);
        expect(wrapper.vm.additionalInfoSelected).toBe(false);
        expect(wrapper.vm.additionalFilesSelected).toBe(false);
    });

    test('selected step 2', () => {
        const wrapper = shallowMount(FormHolderHeader, { propsData: { ...getData(1) } });
        expect(wrapper.vm.personalInfoSelected).toBe(false);
        expect(wrapper.vm.additionalInfoSelected).toBe(true);
        expect(wrapper.vm.additionalFilesSelected).toBe(false);
    });

    test('selected step 3', () => {
        const wrapper = shallowMount(FormHolderHeader, { propsData: { ...getData(2) } });
        expect(wrapper.vm.personalInfoSelected).toBe(false);
        expect(wrapper.vm.additionalInfoSelected).toBe(false);
        expect(wrapper.vm.additionalFilesSelected).toBe(true);
    });

    test('check onStepSelect with availableStep', () => {
        const wrapper = shallowMount(FormHolderHeader, { propsData: { ...getData(0, 20, 50, 100, [0, 1, 2]) } });

        wrapper.vm.onStepSelect(0);
        expect(wrapper.emitted()['step-selected']).toBeTruthy();

        wrapper.vm.onStepSelect(1);
        expect(wrapper.emitted()['step-selected']).toBeTruthy();

        wrapper.vm.onStepSelect(2);
        expect(wrapper.emitted()['step-selected']).toBeTruthy();
    });

    test('check onStepSelect with unavailableSteps', () => {
        const wrapper = shallowMount(FormHolderHeader, { propsData: { ...getData(0, 20, 50, 100, []) } });

        wrapper.vm.onStepSelect(0);
        expect(wrapper.emitted()['step-selected']).not.toBeTruthy();

        wrapper.vm.onStepSelect(1);
        expect(wrapper.emitted()['step-selected']).not.toBeTruthy();

        wrapper.vm.onStepSelect(2);
        expect(wrapper.emitted()['step-selected']).not.toBeTruthy();
    });
});
