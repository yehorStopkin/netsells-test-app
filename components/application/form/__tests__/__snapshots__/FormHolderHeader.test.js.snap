// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormHolderHeader component default 1`] = `
<div
  class="form-holder-header"
>
  <application-step-stub
    class="form-holder-header__application-step"
    progress="0"
    selected="true"
    step="0"
    title="STEP 1"
  />
   
  <application-step-stub
    class="form-holder-header__application-step"
    progress="0"
    step="1"
    title="STEP 2"
  />
   
  <application-step-stub
    class="form-holder-header__application-step"
    progress="0"
    step="2"
    title="STEP 3"
  />
</div>
`;

exports[`FormHolderHeader component different filing percentage 1`] = `
<div
  class="form-holder-header"
>
  <application-step-stub
    class="form-holder-header__application-step"
    progress="20"
    selected="true"
    step="0"
    title="STEP 1"
  />
   
  <application-step-stub
    class="form-holder-header__application-step"
    progress="50"
    step="1"
    title="STEP 2"
  />
   
  <application-step-stub
    class="form-holder-header__application-step"
    progress="100"
    step="2"
    title="STEP 3"
  />
</div>
`;

exports[`FormHolderHeader component selected step 1 1`] = `
<div
  class="form-holder-header"
>
  <application-step-stub
    class="form-holder-header__application-step"
    progress="0"
    step="0"
    title="STEP 1"
  />
   
  <application-step-stub
    class="form-holder-header__application-step"
    progress="0"
    selected="true"
    step="1"
    title="STEP 2"
  />
   
  <application-step-stub
    class="form-holder-header__application-step"
    progress="0"
    step="2"
    title="STEP 3"
  />
</div>
`;
